### PR TITLE
Qoqo/Roqoqo Qiskit Devices

### PR DESCRIPTION
**Pull request by mlodi-hqs**
Two new packages containing Qoqo device interfaces for the following Qiskit's backends :
- Belem
- Jakarta
- Lagos
- Lima
- Manila
- Nairobi
- Perth
- Quito

`roqoqo_qiskit_devices` is a full Rust package, `qoqo_qiskit_devices` is the equivalent Python package built via pyo3.
The `device_property` module updates an ibm_device instance with gate times as well as decoherence information via the `qiskit_ibm_provider` package.

Also contains updates to `qoqo_qiskit`'s documentation.

### CODE REVIEW

File: roqoqo_qiskit_devices/src/devices/ibm_belem.rs [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+            });
+        }
+        if target >= self.number_qubits {
+            return Err(RoqoqoError::GenericError {
+                msg: format!(
+                    "Qubit {} larger than number qubits {}",
+                    target, self.number_qubits
+                ),
+            });
+        }
```
Comment (nfwvogt) [inline roqoqo_qiskit_devices/src/devices/ibm_belem.rs:165]: Here we should also add a check that only allows setting the gate time, if the edge is already in the device. So we can prevent the user adding 2qubit connections that are not really available on the physical hardware. This also goes for other devices.

**Review by nfwvogt — CHANGES_REQUESTED** [review]
Some smaller changes have been requested. Also besides the DEPENDENCIES file we should also have the LICENSE_FOR_BINARY_DISTRIBUTION and the Python license files added before we build and publish the wheels.

**Review by nfwvogt — CHANGES_REQUESTED** [review]
Some smaller changes have been requested. Also besides the DEPENDENCIES file we should also have the LICENSE_FOR_BINARY_DISTRIBUTION and the Python license files added before we build and publish the wheels.

File: qoqo_qiskit_devices/docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Welcome to qoqo_qiskit_devices's documentation!
+======================================
+qoqo_qiskit_devices is the interface for Qiskit's backend devices for HQS Quantum Simulation's qoqo package.
+
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Contents:
+
+   modules
```
[the remark this answers is not in the document]
Comment (mlodi-hqs) [reply qoqo_qiskit_devices/docs/index.rst:10]: Isn't it already that one? I don't see the difference with the structure of other packages.

File: qoqo_qiskit_devices/qoqo_qiskit_devices/LICENSE_FOR_BINARY_DISTRIBUTION [hunk 3]
```diff
@@ -0,0 +1,210 @@
+The source of this software is licensed under the Apache 2.0 license (see below)
+In building the binary distribution other products the software listed in the file DEPENDENCIES are also used and/or linked.
+The licenses an Copyright notices are also listed in the DEPENDENCIES file.
+
+This product in binary form links dynamically against Python  available under the Python license. For full license text see file qoqo_calculator_pyo3/PYTHON_LICENSE
```
Comment (nfwvogt) [inline qoqo_qiskit_devices/qoqo_qiskit_devices/LICENSE_FOR_BINARY_DISTRIBUTION:5]: Minor thing, we should replace `qoqo_calculator_pyo3` with `qoqo_qiskit_devices`

**Review by nfwvogt — APPROVED** [review]
Looks good with one minor change

**Pull request merged**